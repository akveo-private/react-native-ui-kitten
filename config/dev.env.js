const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
