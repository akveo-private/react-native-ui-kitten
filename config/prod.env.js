const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
