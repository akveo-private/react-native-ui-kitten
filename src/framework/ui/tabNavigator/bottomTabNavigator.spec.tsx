import React from 'react';
import {
  Image,
  ImageSourcePropType,
} from 'react-native';
import {
  fireEvent,
  render,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
  StyleType,
} from '@kitten/theme';
import * as config from './bottomTabNavigator.spec.config';
import * as tabsConfig from './bottomNavigatorTab.spec.config';
import {
  BottomTabNavigator as BottomTabNavigatorComponent,
  Props as BottomTabNavigatorProps,
} from './bottomTabNavigator.component';
import {
  BottomNavigatorTab as BottomNavigatorTabComponent,
  Props as BottomNavigatorTabProps,
} from './bottomNavigatorTab.component';

const BottomTabNavigator = styled<BottomTabNavigatorComponent, BottomTabNavigatorProps>(BottomTabNavigatorComponent);
const BottomNavigatorTab = styled<BottomNavigatorTabComponent, BottomNavigatorTabProps>(BottomNavigatorTabComponent);

describe('@bottom-tab-navigator: component checks', () => {

  const Mock = (props?: BottomTabNavigatorProps): React.ReactElement<StyleProviderProps> => (
    <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
      <BottomTabNavigator {...props} />
    </StyleProvider>
  );

  const MockTab = (props: BottomNavigatorTabProps): React.ReactElement<StyleProviderProps> => (
    <StyleProvider mapping={tabsConfig.mapping} theme={tabsConfig.theme} styles={{}}>
      <BottomNavigatorTab {...props} />
    </StyleProvider>
  );

  const tab1Uri: string = 'https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png';
  const tab2Uri: string = 'https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png';
  const tab3Uri: string = 'https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png';

  const getTab1Uri = (): ImageSourcePropType => ({ uri: tab1Uri });
  const getTab2Uri = (): ImageSourcePropType => ({ uri: tab2Uri });
  const getTab3Uri = (): ImageSourcePropType => ({ uri: tab3Uri });

  const tabTestId: string = '@tab/last';

  it('* empty', () => {
    const component = render(<Mock children={[]}/>);
    expect(component).toMatchSnapshot();
  });

  it('* with routes', () => {
    const component = render(
      <Mock>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab1Uri()} style={style}/>}
          title='Screen 1'
          selected={false}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab2Uri()} style={style}/>}
          title='Screen 2'
          selected={true}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab3Uri()} style={style}/>}
          title='Screen 3'
          selected={false}/>
      </Mock>,
    );
    expect(component).toMatchSnapshot();
  });

  it('* current index', () => {
    const component =
      <Mock selectedIndex={1}>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab1Uri()} style={style}/>}
          title='Screen 1'
          selected={false}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab2Uri()} style={style}/>}
          title='Screen 2'
          selected={true}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab3Uri()} style={style}/>}
          title='Screen 3'
          selected={false}/>
      </Mock>;
    const rendered = render(component);
    expect(component.props.selectedIndex).toBe(1);
    expect(rendered).toMatchSnapshot();
  });

  it('* tab choose', () => {
    const onSelect = jest.fn();
    const component = render(
      <Mock
        selectedIndex={1}
        onSelect={onSelect}>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab1Uri()} style={style}/>}
          title='Screen 1'
          selected={false}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab2Uri()} style={style}/>}
          title='Screen 2'
          selected={true}/>
        <MockTab
          testID={tabTestId}
          icon={(style: StyleType) => <Image source={getTab3Uri()} style={style}/>}
          title='Screen 3'
          selected={false}/>
      </Mock>,
    );

    fireEvent(component.getByTestId(tabTestId), 'select');

    expect(component).toMatchSnapshot();
    expect(onSelect).toHaveBeenCalled();
  });

  it('* choose selected tab', () => {
    const onSelect = jest.fn();
    const component = render(
      <Mock
        selectedIndex={1}
        onSelect={onSelect}>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab1Uri()} style={style}/>}
          title='Screen 1'
          selected={false}/>
        <MockTab
          testID={tabTestId}
          icon={(style: StyleType) => <Image source={getTab2Uri()} style={style}/>}
          title='Screen 2'
          selected={true}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab3Uri()} style={style}/>}
          title='Screen 3'
          selected={false}/>
      </Mock>,
    );
    fireEvent(component.getByTestId(tabTestId), 'select');
    expect(component).toMatchSnapshot();
    expect(onSelect).not.toHaveBeenCalled();
  });

  it('* additional appearance', () => {
    const component = render(
      <Mock appearance='highlight'>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab1Uri()} style={style}/>}
          title='Screen 1'
          selected={false}/>
        <MockTab
          icon={(style: StyleType) => <Image source={getTab2Uri()} style={style}/>}
          title='Screen 2'
          selected={true}/>
        <MockTab
          testID={tabTestId}
          icon={(style: StyleType) => <Image source={getTab3Uri()} style={style}/>}
          title='Screen 3'
          selected={false}/>
      </Mock>,
    );
    expect(component).toMatchSnapshot();
  });

});

