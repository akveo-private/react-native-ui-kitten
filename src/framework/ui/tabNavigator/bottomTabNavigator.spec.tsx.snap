// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@bottom-tab-navigator: component checks * additional appearance 1`] = `
<View
  appearance="highlight"
  dispatch={[Function]}
  selectedIndex={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={0}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#3366FF",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
          "fontWeight": "600",
        }
      }
    >
      Screen 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 2
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="@tab/last"
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * choose selected tab 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  onSelect={[MockFunction]}
  selectedIndex={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={1}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="@tab/last"
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#3366FF",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
          "fontWeight": "600",
        }
      }
    >
      Screen 2
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * current index 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  selectedIndex={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={1}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#3366FF",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
          "fontWeight": "600",
        }
      }
    >
      Screen 2
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * empty 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  selectedIndex={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={0}
    selectedPosition={0}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "Infinity%",
      }
    }
  />
</View>
`;

exports[`@bottom-tab-navigator: component checks * tab choose 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  onSelect={
    [MockFunction] {
      "calls": Array [
        Array [
          2,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  selectedIndex={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={1}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#3366FF",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
          "fontWeight": "600",
        }
      }
    >
      Screen 2
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="@tab/last"
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * with routes 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  selectedIndex={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={0}
    style={
      Object {
        "0": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "1": Object {
          "position": "absolute",
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/examples.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#3366FF",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
          "fontWeight": "600",
        }
      }
    >
      Screen 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/attachment.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 2
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn0.iconfinder.com/data/icons/customicondesignoffice5/256/announcements.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": undefined,
          "tintColor": "#8992A3",
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
          "fontWeight": "600",
        }
      }
    >
      Screen 3
    </Text>
  </View>
</View>
`;
